rewrite-java/src/main/java/org/openrewrite/java/TreeVisitingPrinter.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Spliterators;
+import java.util.stream.Collectors;
+import org.openrewrite.Cursor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.InMemoryExecutionContext;
+import org.openrewrite.Tree;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.tree.J;
+
+import static java.util.stream.StreamSupport.*;
+
+
+/**
+ * A visitor to print a tree visiting order and hierarchy in format like below.
+ * <pre>
+ * #root
+ * \---J.CompilationUnit | "..."
+ *     |---#JRightPadded
```

Comment:
Padding elements like `JRightPadded` are very common throughout the tree. True to the name, they keep track of whitespace may appear to the right of the AST element they enclose. For getting a sense of the structure of the tree these can be skipped over.

Reply:
It makes sense to me, by removing the `JRightPadded` and `JLeftPadded` from the output, the example output tree print will reduce from 107 lines to 70 lines, let me make an update to skip over `JRightPadded` and `JLeftPadded` in the output.

Reply:
skipped unvisited elements like `JRightPadded` or `JLeftPadded` from the output, but keeps an option to print all elements which may be extendable in the future.